monitors/didyoumean.js

```diff
@@ -1,24 +1,33 @@
+/* eslint-disable no-use-before-define, no-restricted-syntax */
+
 const levenshtein = require("fast-levenshtein");
 
+exports.init = (client) => {
+  this.minDist = client.config.minDist && Number.isInteger(client.config.minDist) && client.config.minDist >= 1 ? client.config.minDist : 1;
+};
+
 exports.run = async (client, msg) => {
-  if (msg.author.bot) return;
-  if (client.config.selfbot && msg.author.id !== client.user.id) return;
+  if (msg.author.bot || (client.config.selfbot && msg.author.id !== client.user.id)) return;
+
+  const { prefixLength, command, prefix } = parseCommand(client, msg);
 
-  const conf = client.configuration.get(msg.guild);
-  const prefixLength = client.funcs.parseCommand(client, msg, true);
   if (!prefixLength) return;
-  const command = client.funcs.parseCommand(client, msg);
-  if (command.length && !(client.commands.has(command) || client.aliases.has(command))) {
-    const distances = [];
-    client.commands.filter(c => c.conf.permLevel <= msg.member.permLevel).forEach((val, cmd) => distances.push({
+  if (!command.length && (client.commands.has(command) || client.aliases.has(command))) return;
+
+  const distances = [];
+
+  for (const cmd in client.commands) {
+    if (!client.funcs.checkPerms(client, msg, cmd.conf.permLevel)) continue;
```

Comment:
This is okay for a selfbot, but that's almost like generating the help command in every single message that starts with the prefix, make an external function able to calculate highest permission level and try again.